Sources/Sentry/NSDate+SentryExtras.m

```diff
@@ -18,14 +18,33 @@ + (NSDateFormatter *)getIso8601Formatter
     return isoFormatter;
 }
 
++ (NSDateFormatter *)getIso8601FormatterWithMillisecondPrecision
+{
+    static NSDateFormatter *isoFormatter = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        isoFormatter = [[NSDateFormatter alloc] init];
+        [isoFormatter setLocale:[NSLocale localeWithLocaleIdentifier:@"en_US_POSIX"]];
+        isoFormatter.timeZone = [NSTimeZone timeZoneWithAbbreviation:@"UTC"];
+        [isoFormatter setDateFormat:@"yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"];
+    });
+
+    return isoFormatter;
+}
+
 + (NSDate *)sentry_fromIso8601String:(NSString *)string
 {
-    return [[self.class getIso8601Formatter] dateFromString:string];
+    NSDate *date = [[self.class getIso8601FormatterWithMillisecondPrecision] dateFromString:string];
+    if (nil == date) {
+        return [[self.class getIso8601Formatter] dateFromString:string];
```

Comment:
`l`: Can you please add a comment here, that we do this to be backward compatible.

Reply:
done